### DISCUSSION TITLE
reveal.js presentation: Slides without title show up in TOC as empty bullet points

### DISCUSSION THREAD

**Discussion by t-klinge** [quarto-dev/quarto-cli#11102] [Q&A]
### Description

Dear all,

I'm working on my first set of slides with Quarto after being deeply impressed by its other functionalities. 

However, I'm facing a similar issue like the one described in [#3671](https://github.com/quarto-dev/quarto-cli/issues/3671): Since I do not want sections represented as endless scrollable slides, I wish to have more than one slide per section. If I create a new slide (without a title) by using `---`, these slides show up as empty bullet points in the TOC (see below). How can I change this so the TOC only shows those slides which have an explicit title? Alternatively, is there a way to have several slides with the same title but only one entry in the TOC?

My apologies if this question has already been answered elsewhere. I tried my best to study the existing answers but couldn't get my head around them.

Many thanks for your help in this matter!

Reproducible example:

````qmd
---
title: "My Title"
format: revealjs
editor: source
toc: true
---
## Section 1
Slide 1 of section 1. Should show up in the TOC as a bullet point.

---

Slide 2 of section 1. Should **not** show up in the TOC as a bullet point.

## Section 2
Slide 1 of section 2. Should show up in the TOC as a bullet point.
````

**Edit**: I tried fiddling with the `{.unlisted}` option in the headings. This seems to work for first-level headings but not for second-level headings.

````qmd
---
title: "My Title"
format: revealjs
editor: source
toc: true
---
## Section 1
Slide 1 of section 1. Should show up in the TOC as a bullet point.

---

## Section 1 {.unlisted}

Slide 2 of section 1. Should **not** show up in the TOC as a bullet point. **Does work.**

---

### Section 1, subsection 1 {.unlisted}

Slide 1 of section 1, subsection 1. Should **not** show up in the TOC as a bullet point. **Does not work.**

## Section 2
Slide 1 of section 2. Should show up in the TOC as a bullet point.
```

**Comment by mcanouil** [comment]
You might want to look into the other related discussions: https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=is%3Aopen+toc+label%3Arevealjs

**Reply by t-klinge** [reply]
Thanks, mcanouil. 
As you say [here](https://github.com/quarto-dev/quarto-cli/discussions/10386#discussioncomment-10158958) , your approach would be to copy `## First Section` to the second slide but hide it using `## First section {.unlisted}`? Just making sure I'm catching your idea as intended. I edited the original post to show my troubles in following this approach for second-level headings.

**Reply by mcanouil** [reply]
It's not "my approach", it's the documented approach.

And no you don't need to copy anything. An empty header is enough.

Note that the menu, TOC, etc. are not Quarto features. They come from Reveal.js. (Discussed in other threads about hiding slides titles)
Looking at Reveal.js documentation can help understand this format.

**Comment by cderv** [comment]
> However, I'm facing a similar issue like the one described in https://github.com/quarto-dev/quarto-cli/issues/3671: Since I do not want sections represented as endless scrollable slides, I wish to have more than one slide per section. If I create a new slide (without a title) by using ---, these slides show up as empty bullet points in the TOC (see below).

I think the fix from #3671 is not working as intended. I'll patch this to fix, and add a test to confirm. 

Empty bullet point in TOC should not be shown - we are supposed to remove them as Pandoc post processing. 

> Alternatively, is there a way to have several slides with the same title but only one entry in the TOC?

the `toc: true` feature is not that clever. It is a way to have TOC in a slide; This is a Pandoc feature for revealjs format. 

The Menu plugin with TOC on the sidebar is more complete, and offers more configuration. It is a Revealjs Feature as a plugin.

**Reply by cderv** [reply]
So in fact this using `---` the problem. 

Use `##` instead for empty slide delimiter and it will works. 

Support for `---` will be fixed in 
- #11146
